app/services/cache/MongoCacheClient.scala

```diff
@@ -214,16 +242,23 @@ class MongoCacheClient(appConfig: ApplicationConfig, db: () => DefaultDB, applic
   /**
     * Fetches the whole cache and returns default where not exists
     */
-  def fetchAllWithDefault(credId: String): Future[Cache] =
+  def fetchAllWithDefault(credId: String): Future[Cache] = {
     fetchAll(Some(credId)).map {
       _.getOrElse(Cache(credId, Map.empty))
     }
+  }
 
   /**
     * Removes the item with the specified id from the cache
     */
-  def removeById(credId: String) =
-    collection.delete().one(key(credId)) map handleWriteResult
+  def removeById(credId: String):Future[Boolean] = {
+    collection.findOneAndDelete(key(credId)).toFuture()
+      .map { result => true
```

Comment:
❓ Should we check for the number of deleted rows? Wouldn't this return true even if there were zero deletions?

Reply:
as mentioned in the previous comment, we could log that if it would be helpful but I think if the goal is to not have something in the db matching that id then either way the code has succeeded

Reply:
In that case, consider switching to returning `Unit` - it would tell you if there are any redundant checks when using this function ;)